projects/graph/src/graph.py

```diff
@@ -1,9 +1,112 @@
 """
 Simple graph implementation compatible with BokehGraph class.
 """
+import random
 
+class Queue:
+    def __init__(self):
+        self.queue = []
+    def enqueue(self, value):
+        self.queue.append(value)
+    def dequeue(self):
+        if (self.size()) > 0:
+            return self.queue.pop(0)
+        else:
+            return None
+    def size(self):
+        return len(self.queue)
+
+class Stack:
+    def __init__(self):
+        self.stack = []
+    def push(self, value):
+        self.stack.append(value)
+    def pop(self):
+        if (self.size()) > 0:
+            return self.stack.pop()
+        else:
+            return None
+    def size(self):
+        return len(self.stack)
+    
 
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices = {}
+    def add_vertex(self, vert_id):
+        self.vertices[vert_id] = Vertex(vert_id)
+    def add_edge(self, vert1, vert2):
+        if vert1 in self.vertices and vert2 in self.vertices:
+            self.vertices[vert1].edges.add(vert2)
+            self.vertices[vert2].edges.add(vert1)
+        else:
+            raise IndexError('Vertex does not exist')
+    def add_directed_edge(self, vert1, vert2):
+        if vert1 in self.vertices:
+            self.vertices[vert1].edges.add(vert2)
+        else:
+            raise IndexError('Vertex does not exist')
+    def dft_st(self, starting_node):
+        s = Stack()
+        s.push(starting_node)
+        visited = []
+        while s.size() > 0:
+            current = s.pop()
+            if current not in visited:
+                visited.append(current)
+                print(visited)
+                for edge in self.vertices[current].edges:
+                    s.push(edge)
+
+    def bft(self, starting_node, passed_color):
+        q = Queue()
+        q.enqueue(starting_node)
+        visited = []
+        while q.size() > 0:
+            current = q.dequeue()
+            if current not in visited:
+                visited.append(current)
+                print(visited)
+                self.vertices[current].color = passed_color
+                for edge in self.vertices[current].edges:
+                    q.enqueue(edge)
+
+    def connected(self):
+        color_count = 0
+        colors = ['yellow', 'red', 'blue', 'orange', 'green', 'purple', 'lime', 'olive', 'gold']
+        for vert in self.vertices:
+            if not self.vertices[vert].color != 'grey':
+                self.bft(vert, colors[color_count])
+                color_count +=1
```

Comment:
I think this is the block you're referring to;
`if not self.vertices[vert].color != 'grey'`
I am still looking at it, but I believe there is something about the double negative that is working unusually.